Sources/MapNode.swift

```diff
@@ -0,0 +1,65 @@
+//
+// StencilSwiftKit
+// Copyright (c) 2017 Olivier Halligon
+// Created by Peter Livesey.
+// MIT Licence
+//
+
+import Stencil
+
+public class MapNode: NodeType {
```

Comment:
I wonder if the name of the variable to store the result of the map should be part of the Map node?

I mean, that node syntax reads strange to be and doesn't seem that natural to use (`{% map array with element set varname %}`, I always wonder what `element` and `varname` are, which is which, etc). And the node doesn't produce any output as a result, storing the result in the context instead, that seems convoluted at first read.

That would be awesome if it would be possible to just use `{% map array with x %}Item {{maploop.counter}} = {{x}}.{% endmap %}` and that would just generate / output… an array. That the user can then just use as-is (let it render), or store in a variable using the `SetNode`.

I know that's not really possible now, as `render` outputs a `String` (which is kinda logical), and making it a Filter instead of a Node won't make sense either because we need to let us pass a whole Stencil tree as the "closure" of that "map" — so `{{ array|map:what_to_put_here }}` wouldn't cut it (except… if `what_to_put_here` was a `CallableNode` from a `MacroNode`, heh!) — but I'm wondering if we couldn't find a more elegant solution that this `{% map array with item set result %}`?

\cc @kylef thoughts?